airflow/providers/amazon/aws/hooks/sagemaker.py

```
@@ -954,3 +954,13 @@ def _list_request(
                 return results
             else:
                 next_token = response["NextToken"]
+
+    def find_processing_job_by_name(self, processing_job_name: str) -> bool:
+        """
+        Query processing job existence by describe_processing_job, rather than iterate all the jobs.
+        """
+        try:
+            res = self.get_conn().describe_processing_job(ProcessingJobName=processing_job_name)
+            return True
+        except ClientError as e:
```

Does `ClientError` guarantee the job does not exist, or do we need to check the status code or something? 404?

---

I prefer fail fast. According to the test, ClientError.response['Error']['Code'] == 'ValidationException', when the job doesn't exist.

---

While this API specifically throws only [ResourceNotFound](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_DescribeProcessingJob.html#API_DescribeProcessingJob_Errors) there are many errors that are common to all sagemaker APIs that could technically be thrown (see the list [here](https://docs.aws.amazon.com/sagemaker/latest/APIReference/CommonErrors.html)). For completeness I think it's worth checking directly for ResourceNotFound. It is also more future proof that way. Just my 2c

---

I use boto3 client to verify the api behavior, it returns validation exception rather than ResourceNotFound